src/main/java/com/ss/utopia/UserController.java

```diff
@@ -43,14 +56,46 @@
 	@GetMapping()
 	public ResponseEntity<Object> findAll() 
 	throws ConnectException, SQLException {
-
+		
 		List<User> userList = userService.findAll();
 		return !userList.isEmpty() 
 		? new ResponseEntity<>(userList, HttpStatus.OK)
 		: new ResponseEntity<>(null, HttpStatus.NO_CONTENT);
 	}
 	
 	
+	cd 
+	public ResponseEntity<?> forgotPassword(@RequestBody LinkedHashMap uMap ) throws ConnectException, IllegalArgumentException, SQLException, UserNotFoundException{
+		
+		String email = (String) uMap.get("email");
+
+		try {
+			userService.sendRecoveryEmail(email);
+			return new ResponseEntity<>("ok", HttpStatus.OK);	
+		} catch (UserNotFoundException err) {
+			return new ResponseEntity<>(err.getMessage(), HttpStatus.NOT_FOUND);
+		}	
+	}
+	
+	@PostMapping("/forgot-password/recover")
+	public ResponseEntity<?> passwordRecovery(@RequestBody LinkedHashMap uMap) throws ExpiredTokenExpception, TokenNotFoundExpection, ConnectException, IllegalArgumentException, SQLException, UserNotFoundException  {
+		
+		String recoveryCode = (String) uMap.get("recoveryCode");
+		String password = (String) uMap.get("password");
+		
+		try {
+			userService.verifyToken(recoveryCode);
+			userService.ChangePassword(userService.verifyToken(recoveryCode), password);
+			return new ResponseEntity<>("Password successfully changed ", HttpStatus.OK);
+			
+		} catch (ExpiredTokenExpception e) {
+			return new ResponseEntity<>(e.getMessage(), HttpStatus.UNAUTHORIZED);
+		} 
+		catch (Exception e) {
+			return new ResponseEntity<>(e.getMessage(), HttpStatus.UNAUTHORIZED);
```

Comment:
If an unexpected Exception is caught, why are we returning a 403 status instead of 500?